spk/syncthing/src/service-setup.sh

```diff
@@ -13,6 +13,13 @@ service_prestart ()
         SERVICE_COMMAND="${SERVICE_COMMAND} ${SYNCTHING_OPTIONS}"
     fi
 
+    # If the system has a TLS certificate for us, force its usage
+    cert_dir=/usr/local/etc/certificate/${SYNOPKG_PKGNAME}/${SERVICE_CERT}
+    if [ -f ${cert_dir}/cert.pem -a -f ${cert_dir}/privkey.pem ]; then
+        ln -sf ${cert_dir}/cert.pem ${SYNOPKG_PKGVAR}/https-cert.pem
+        ln -sf ${cert_dir}/privkey.pem ${SYNOPKG_PKGVAR}/https-key.pem
+    fi
+
```

Comment:
Can we make this optional, i.e. activate with an option in the install wizard?

When using a web proxy with dyndns not hosted on the diskstation we don't want to use such certificates for TLS.
Probably a Checkbox in the Wizard like "[ ] Activate TLS with DSM integrated Certificate".
This would also allow to abort here with an error message, when certificate is not available.

Reply:
I like that idea, but not sure if it will work in all cases.  The `ca_reloader` script will be called whenever the *configured* certificate changes AFAIU.  By default, the "system" certificate is assigned, meaning the synology.com-derived certificate unless the user changed it.  Now if that one is automatically renewed by DSM, the ca_reloader script will replace Syncthing's self-signed certificate, regardless of whether the option was checked during installation.

There's no easy way to store such wizard settings permanently for later use, or is there?

Could you elaborate on the case where you think relying on Syncthing's own self-signed cert is the only solution?

Reply:
Ping @hgy59.  Please revisit all open review comments here, there are some remaining questions.